src/services/fetchers/CryptoComparePriceWSFetcher.ts

```diff
@@ -2,19 +2,20 @@ import {inject, injectable} from 'inversify';
 
 import CryptoCompareWSClient from '../ws/CryptoCompareWSClient.js';
 import {PairWithFreshness} from '../../types/Feed.js';
+import {FeedBaseQuote, FeedFetcherInterface} from 'src/types/fetchers.js';
 
 @injectable()
-class CryptoComparePriceWSFetcher {
+class CryptoComparePriceWSFetcher implements FeedFetcherInterface {
   @inject(CryptoCompareWSClient) cryptoCompareWSClient!: CryptoCompareWSClient;
 
-  async apply(pair: PairWithFreshness, timestamp: number): Promise<number> {
-    const price = await this.cryptoCompareWSClient.getLatestPrice(pair, timestamp);
+  async apply(params: PairWithFreshness & FeedBaseQuote, timestamp: number): Promise<number> {
```

Comment:
isn't it better to simply add additional param (might be optional) or with optional fields eq 
```
(..., options?: FetcherOptions)

FetcherOptions {
base?: string,
quote?: string
}
```

this will allow to add easily anything in future

this should be as part of general interface

Reply:
I agree, this also will simplify the complex line you mentioned above.

The only drawback I can see is with this is that it becomes a bit confusing what the `params` are in the end. This is are in the end the custom params of the fetcher (those that corresponds to each particular fetcher). Maybe just changing the name to `customParams` may help.

Reply:
`params` will be dedicated to fetcher only, oether options will be generic, same for all fetchers

Reply:
Dariousz, wouldn't be an option to pass base & quote as mandatory (not optional) values to the `apply` method. Like:

```typescript
export interface FeedFetcherInterface {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  apply(params: any, base: string, quote: string, timestamp?: number): FeedFetcherInterfaceResult;
}
```

Since this values are mandatory for the model I think it makes this method more predictive in behavior if we all the time pass something to it for thos values. What do you prefer?

Reply:
1. I don't want ot use `any` (but this might be separate PR just for `param` types
2. if the symbol is already easy to convert to base-quote I don't want to duplicate this info, 99% of cases will like this, so by default we will be able to handle 99% of cases out of the box, no changes needed, so I prefer to have this optional

Reply:
3. if we keep `params` to be just dedicated params for fetcher it is easier to work with it inside fetcher and easy to see what is expected in config

Reply:
It sounds reasonable. I can make it optional.

My only concern is that the symbol is converted to base & quote before this call. In this way, the fetcher dedicates entirely to fetch prices and does not try to compute base & quote. If this is the case, base & quote shouldn't be optional. If they cannot be computed then makes no sense to call the fetcher (this is how I thought it originally)

Reply:
ah I see now your point, so:
1. base and quote will be optional in yaml config file
2. base and quote will be required in `options` in fetcher arguments

I hope now we on the same page.

so as you said, if we can not calculate base-quote we will not fetch anything.

Reply:
Now we are on the same page :)

Let me move from there.